clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -19,56 +19,91 @@
+import org.apache.kafka.common.requests.ConsumerGroupHeartbeatRequest;
+import org.apache.kafka.common.utils.LogContext;
+import org.slf4j.Logger;
 
 import java.util.Optional;
 
 /**
- * Membership manager that maintains group membership for a single member following the new
+ * Membership manager that maintains group membership for a single member, following the new
  * consumer group protocol.
  * <p/>
- * This keeps membership state and assignment updated in-memory, based on the heartbeat responses
- * the member receives. It is also responsible for computing assignment for the group based on
- * the metadata, if the member has been selected by the broker to do so.
+ * This is responsible for:
+ * <li>Keeping member info (ex. member id, member epoch, assignment, etc.)</li>
+ * <li>Keeping member state as defined in {@link MemberState}.</li>
+ * <p/>
+ * Member info and state are updated based on the heartbeat responses the member receives.
  */
 public class MembershipManagerImpl implements MembershipManager {
 
+    /**
+     * ID of the consumer group the member will be part of., provided when creating the current
+     * membership manager.
+     */
     private final String groupId;
+
+    /**
+     * Group instance ID to be used by the member, provided when creating the current membership manager.
+     */
     private final Optional<String> groupInstanceId;
+
+    /**
+     * Member ID assigned by the server to the member, received in a heartbeat response when
+     * joining the group specified in {@link #groupId}
+     */
     private String memberId;
+
+    /**
+     * Current epoch of the member. It will be set to 0 by the member, and provided to the server
+     * on the heartbeat request, to join the group. It will be then maintained by the server,
+     * incremented as the member reconciles and acknowledges the assignments it receives.
+     */
     private int memberEpoch;
+
+    /**
+     * Current state of this member a part of the consumer group, as defined in {@link MemberState}
+     */
     private MemberState state;
+
+    /**
+     * Assignor type selection for the member. If non-null, the member will send its selection to
+     * the server on the {@link ConsumerGroupHeartbeatRequest}. If null, the server will select a
+     * default assignor for the member, which the member does not need to track.
+     */
     private AssignorSelection assignorSelection;
 
     /**
      * Assignment that the member received from the server and successfully processed.
      */
     private ConsumerGroupHeartbeatResponseData.Assignment currentAssignment;
+
     /**
      * Assignment that the member received from the server but hasn't completely processed
      * yet.
      */
     private Optional<ConsumerGroupHeartbeatResponseData.Assignment> targetAssignment;
+
     /**
-     * Latest assignment that the member received from the server while a {@link #targetAssignment}
-     * was in process.
+     * slf4j logger.
```

Comment:
I think this is extraneous?

Reply:
Oh, maybe not. I don't think we need to provide JavaDoc for the loggers, do we?

Reply:
I think it is still helpful, I added it intentionally to be consistent with java docs in all vars (the loggers do have doc in multiple other classes btw)

Reply:
I agree with keeping the javadoc for consistency but we could remove `slf4j`.